iPetCare/Application/Services/InvitationService.cs

```diff
@@ -0,0 +1,156 @@
+
+            Context.Remove(invitation);
+
+            if (await Context.SaveChangesAsync() <= 0)
+                return new ServiceResponse(HttpStatusCode.BadRequest, "Wystąpił błąd podczas usuwania zaproszenia");
+
+            return new ServiceResponse(HttpStatusCode.OK);
+        }
+
+        public async Task<ServiceResponse<ChangeStatusInvitationDtoResponse>> ChangeInvitationStatusAsync(ChangeStatusInvitationDtoRequest dto, Guid invitationId)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var invitation = await Context.Requests.FindAsync(invitationId);
+
+            if (invitation == null)
+                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.NotFound);
+
+            var pet = await Context.Pets.FindAsync(invitation.PetId);
+
+            if (CurrentlyLoggedUser.Id == invitation.UserId && pet.OwnerPets.Where(x => x.Owner.User.Id == CurrentlyLoggedUser.Id).Any())
+                return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.Forbidden);
+
+            invitation.IsAccepted = dto.IsAccepted;
+
+            var responseDto = Mapper.Map<ChangeStatusInvitationDtoResponse>(invitation);
+
+            if (dto.IsAccepted)
+            {
+                var User = UserManager.FindByIdAsync(invitation.UserId).Result;
+
+                if (User.Role == Role.Owner)
+                {
+                    var owner = await Context.Owners.Where(x => x.UserId == invitation.UserId).SingleOrDefaultAsync();
+
+                    var existInvitation = await Context.OwnerPets.Where(x => x.PetId == invitation.PetId && x.OwnerId == owner.Id).ToListAsync();
+                    if (existInvitation.Any())
+                        return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.BadRequest, "Podany użytkownik został zatwierdzony do tego zwierzaka");
+
+                    Context.OwnerPets.Add(new OwnerPet
+                    {
+                        Pet = pet,
+                        Owner = owner
+                    });
+                }
+
+                if (User.Role == Role.Vet)
+                {
+                    var vet = await Context.Vets.Where(x => x.UserId == invitation.UserId).SingleOrDefaultAsync();
+
+                    var existInvitation = await Context.VetPets.Where(x => x.PetId == invitation.PetId && x.VetId == vet.Id).AnyAsync();
+                    if (existInvitation)
+                        return new ServiceResponse<ChangeStatusInvitationDtoResponse>(HttpStatusCode.BadRequest, "Podany użytkownik został zatwierdzony do tego zwierzaka");
+
+                    Context.VetPets.Add(new VetPet
+                    {
+                        Pet = pet,
+                        Vet = vet
+                    });
+                }
+            }
+
+            if (dto.IsAccepted == false)
```

Comment:
we no daj !dto.IsAccepted. Oszczędzimy 8 znaków!

(A na pisanie tego komentarza zmarnowałem ponad 20 znaków :C )

Reply:
Okej :C